### DISCUSSION TITLE
DIN Rail Mounting

### DISCUSSION THREAD

**Discussion by EdsgnDesigner** [phil-barrett/grblHAL-teensy-4.x#79] [Ideas]
![Screenshot_20230630-224452](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/85149739/52a6c65e-0a88-40d1-a60f-0e1cb7a02fb4)
Phil, What are the chances that you could make the board edges slightly different so that it could be mounted in something like this? Readily available on eBay. I do alot of commercial/ industrial builds and would love to mount one of your boards into something like this.......

**Comment by phil-barrett** [comment]
Yes, I think din rail mounting is a good idea.  I have done a couple of smaller PCBs, a relay board for example, that are set up for din rail mount though I think smaller than the one you are showing.

I am looking at a spin of the T41 board and a PICO based board.  I think there may be a way to put din rail mount holes on it. What is the one in the photo - do you have a link?

**Comment by EdsgnDesigner** [comment]
Hi Phil, Thanks for the reply. Here is the link I was looking at.

https://www.ebay.ca/itm/254820158374?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=a0OmVZrhTVy&sssrc=2349624&ssuid=vLUg76iqQ7q&var=554650287643&widget_ver=artemis&media=Copy 

Are you thinking some kind of stand alone/ offline controller with the pico board?

**Reply by phil-barrett** [reply]
My boards are typically 92x94.5 mm.  This is driven by the need to have the screw terminals on the edge. It may be possible to adapt to the UM90 spec.  I believe that means a 90mm PCB width - is that correct?  I had not looked at that carrier design before.  I was thinking of setting it up for mounts similar this one:
![image](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/16570958/e8b186e2-d180-40e4-99fe-3b60a180b73a)
Two mounts about 15-20mm from each side of the board. Not sure if the board would be supported well enough though - it takes a fair amount of movement when wiring it in. The tray approach looks more substantial.

How do you attach the PCB to the UM90 tray?  Clamp?  Screws? I've looked at several sites and still don't have a clear understanding of it.  Maybe I'll just order a couple different ones to get more familiar.

As to the PICO, no plans for a true stand alone controller.  It does support the PICO W.  I am looking at adding a microSD socket which would allow use of the grblHAL WEB UI.  [edit: that is probably confusing.  the microSD would allow storing the GCode files and WEB UI would allow running them from there]

**Comment by EdsgnDesigner** [comment]
Yes the board would have to be 90 mm tall... or something completely different. These come in a variety of sizes. The board just slides into the middle part and then held in place with the end caps.

Also, Have you made your board files available?
My plan is to use two teensy boards connected to each other. 1 will run grblHal and the other will act as a mediator between an industrial PLC via Modus Ethernet and the grblHal board connected to the hardware. I have done some boards in the past but nothing as complicated as yours. I was just wondering if it was posted somewhere I could use it as a starting point? 

![Screenshot_20230701-160826](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/85149739/95e5b70b-58c9-42cb-973c-8d67519960ab)

![Screenshot_20230701-161159](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/85149739/064a9165-3b60-420d-a0fa-0aa2f5df8c2d)

![Screenshot_20230701-160859~2](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/85149739/0e313e28-e643-4572-b98b-d989a8921963)

**Comment by phil-barrett** [comment]
Hmmm, it's not clear to me how high the endcaps are - they can't be more than 0.8mm above the top of the PCB or they will block the screw terminals. I will need to pull the screw terminals back from the edge by a bit - maybe 2 mm, will need to order one to measure that.

To be honest, I don't want to give away the PCB files. Perhaps I will open source them some day but for now, I just open source the schematics.

**Comment by phil-barrett** [comment]
I just ordered a couple on Aliexpress.  The ebay links also come from china so it's probably a toss-up who's faster.

Thanks for dropping me a note on this. 

[Here's what I ordered](https://www.aliexpress.us/item/2251832441075383.html?spm=a2g0o.order_list.order_list_main.5.26271802NHTXPt&gatewayAdapt=glo2usa)

**Comment by EdsgnDesigner** [comment]
No problem. I will keep an eye out in case you release a board in the din mount version. For now I will just use the holes to mount it to a backer board. No worries about not releasing the board files. I will just do a separate controller board and mount them close to each other for now.
